src/activities/quizzes/timing/QuizTimingEntity.js

```diff
@@ -38,8 +38,17 @@ export class QuizTimingEntity extends Entity {
 		return entity && entity.hasActionByName(Actions.quizzes.timing.updateTimeLimit);
 	}
 
+	canEditTimingTimeLimit() {
+		const entity = this.getTimerSettingsSubEntity();
+		return !!entity?.hasActionByName(Actions.quizzes.timing.updateTimeLimit);
```

Comment:
can we follow the syntax above instead of using double bang?

Reply:
I just tried using the above syntax. It seems that this breaks the test case where I want to test that entities with the flag off structure should expect this to be false. Without the double bang, this will just return undefined (I guess the difference with this one is that `entity` just doesn't exist in flag off structure, but for flag off functions on can edit x, a read-only version of recommended & enforced subentities still exist in the structure. I could try to expect change this (and other blocks with `!!`) to something like:
```
if (!entity) return false;
return entity.hasActionByName(Actions.quizzes.timing.updateTimeLimit);
```
if we don't want double bang?

Reply:
I think in general we just do `if (!entity) return;` if we expect entity to be undefined

It's fine~ You could just keep it as is 😃